### ISSUE TITLE
Highlight language is always empty

### ISSUE THREAD

**Issue by panda-z** [gohugoio/hugo#6669]
### What version of Hugo are you using (`hugo version`)?

<pre>
$ hugo version
Hugo Static Site Generator v0.62.0/extended linux/amd64 BuildDate: unknown
</pre>

### Does this issue reproduce with the latest release?
Yes

https://github.com/gohugoio/hugo/blob/master/markup/goldmark/convert.go#L297

In this line, `hasLang` is always empty.

E.g., expected HTML output is ```<code class="language-toml" data-lang="toml">```.
However, it current outputs ```<code>```.

**Comment by thetanil** [comment]
I'm not a Go programmer, but that code looks OK to me. I tested, but was unable to reproduce this issue. using 0.62/extended on win:

<pre>
<code class="language-toml" data-lang="toml">
</pre>

**Comment by panda-z** [comment]
My bad. I didn't mention I've set "linenos=table". And the ```<code>``` I mean is the wrapper of line numbers.

Use previous versions of Hugo, the output is something like

```
<pre class="chroma"><code class="language-toml" data-lang="toml"><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span></code></pre>
```

In the newest version, the output is something like

```
<pre class="chroma"><code><span class="lnt"> 1
</span><span class="lnt"> 2
</span><span class="lnt"> 3
</span></code></pre>
```

I don't know which is the correct one.

**Comment by thetanil** [comment]
Dug in a little. Here's what I got:

### some markdown:

{{< highlight toml "linenos=table" >}}
here:
  is:
    some: 'toml'
{{< /highlight >}}

_________________________________

### in 0.62 we get:

```html
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-toml" data-lang="toml"><span class="nx">here</span><span class="err">:</span>
  <span class="nx">is</span><span class="err">:</span>
    <span class="nx">some</span><span class="err">:</span> <span class="s1">&#39;toml&#39;</span></code></pre></td></tr></table>
</div>
</div>
```

_________________________

### in 0.59 we had:

```html
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code class="language-toml" data-lang="toml"><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-toml" data-lang="toml"><span class="nx">here</span><span class="err">:</span>
  <span class="nx">is</span><span class="err">:</span>
    <span class="nx">some</span><span class="err">:</span> <span class="s1">&#39;toml&#39;</span></code></pre></td></tr></table>
</div>
</div>
```

**Comment by thetanil** [comment]
So the class for the language is not set on the table's number column which totally makes sense to me. But it is different.

I didn't see that the commonmark spec has anything to say about this at all. So what is "right" is maybe not defined. I assume this is related to the upgrade to goldmark, but I don't know if hugo passes to highlight, or if commonmark passes to highlight and this whole thing is upstream. Maybe @bep can comment.

**Comment by satotake** [comment]
Fixed after https://github.com/gohugoio/hugo/pull/6901 ,
released as [v0.65.0](https://github.com/gohugoio/hugo/releases/tag/v0.65.0) .
I tested that @thetanil 's markdown is rendered correctly with Hugo v0.65.2. 
Why do not you close this? @bep

**Comment by panda-z** [comment]
@satotake I've tested with 0.65.2 and the issue still exists.

**Comment by satotake** [comment]
@panda-z hm...,  Let me check again.
Can you show me your config?

**Comment by satotake** [comment]
@panda-z 

I created the [repo](https://github.com/satotake/hugo-issue-6669) for checking.

#### Deploy results
|Hugo version | Shortcode | CodeFence |
|--|--|--|
|[0.54.0](https://v0540--trusting-shockley-8136a8.netlify.com/posts/code/)| OK | NG |
|[0.59.1](https://v0591--trusting-shockley-8136a8.netlify.com/posts/code/)| OK | NG |
| [0.62.0](https://5e51daded5fc90000b571ea4--trusting-shockley-8136a8.netlify.com/posts/code/) | OK | OK | 
| [0.65.2](https://v0652--trusting-shockley-8136a8.netlify.com/posts/code/) | OK | OK |

In my settings, both v0.62.0 and 0.65.2 works fine.
So we have to know a little bit more details of yours if you do not mind...

**Comment by panda-z** [comment]
@satotake 

Could you set `pygmentsUseClasses = true` and search for  `<pre class="chroma"><code`?

The `class="language-toml"` property for `<code>` of line numbers is missing.

**Comment by satotake** [comment]
@panda-z 

Thank you. At last, I got what you mean.

it will make you sad but [pygments was removed](https://github.com/gohugoio/hugo/issues/4491) 
we are migrating into `chroma`.

Related issue: https://github.com/gohugoio/hugo/issues/6577

The class name you pointed here can be tweaked but pygments-related configurations will be removed sooner or later.

Therefore I recommend that you also highlight with `chroma`,
Could you try [it](https://gohugo.io/getting-started/configuration-markup#highlight)?
It is based on `pygments`' concepts, so it will be acceptable for you.

**Comment by panda-z** [comment]
@satotake 

OK. So pygements related configuration still exists only for backward compatibility. They will finally be removed in some future version. That makes sense.

I've tried `noClasses = false` with chroma, but the generated content is the same as that of `pygmentsUseClasses = true`.

**Comment by satotake** [comment]
@panda-z 

When you use `chroma` with `noClasses` option, you need to [regenerate syntax.css](https://gohugo.io/content-management/syntax-highlighting/#generate-syntax-highlighter-css) and load it.

As long as you combine `chroma` with the generated css, it works without `class="language-toml"` of line numbers.
Because, generally speaking, styles of line numbers are not dependent of language as @thetanil suggested.

Or, would you like to change styles of line numbers between languages?
If so, how do you think about writing some css for your purpose instead of hugo's (or maybe upstream) internal implementation

**Comment by panda-z** [comment]
@satotake 

I have generated syntax.css and added it to the CSS file.

Yes, you are right. The styles of line numbers are independent of language.

The problem is that the theme I used selects the line numbers element in order to add language type (e.g., 'toml' for TOML snippet) for each code block.
The language type depends on the attribute (e.g., `class="language-toml"`).
Without it, the theme can not set language type properly. 

Anyway, I've found a solution that uses JS to add the desired attribute.
